src/USER-CONP/fix_charge_update.cpp

```diff
@@ -377,23 +395,144 @@ std::vector<int> FixChargeUpdate::local_to_matrix() {
 /* ---------------------------------------------------------------------- */
 
 void FixChargeUpdate::pre_force(int) {
+  update_charges();
+  short_range_correction(true);
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixChargeUpdate::update_charges() {
+  int *mask = atom->mask;
   std::vector<int> mpos = local_to_matrix();
-  int const nmax = atom->nmax;
+  int const nall = atom->nlocal + atom->nghost;
   vector_compute->compute_vector();
   double *b = vector_compute->vector;
-  for (int i = 0; i < nmax; i++) {
-    int const pos = mpos[i];
-    if (pos < 0) continue;
+  for (int i = 0; i < nall; i++) {
+    if (!(groupbit & mask[i])) continue;
     double q_tmp = 0;
     for (int j = 0; j < ngroup; j++) {
-      q_tmp += elastance[pos][j] * (psi[j] - b[j]);
+      q_tmp += elastance[mpos[i]][j] * (psi[j] - b[j]);
     }
     atom->q[i] = q_tmp;
   }
 }
 
 /* ---------------------------------------------------------------------- */
 
+double FixChargeUpdate::compute_scalar() {
+  return electrode_energy(local_to_matrix()) + short_range_correction(false);
```

Comment:
Für short_range_correction() vielleicht auch eine Überladung? Also ohne True gibt er einfach nur die Energien zurück?

Reply:
Ich finde es gut wenn man den Wert setzen muss. Dann vergisst man es auch nicht so leicht.
Können wir aber machen, wenn dir das wichtig ist.

Reply:
Vielleicht so machen wie in pair->single() mit den Kräften als übergebenen Pointer?